internal/listener/listener.go

```diff
@@ -144,378 +143,13 @@ func (l *Listener) ProcessEvent(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	// TODO: better move all this logic somewhere into incident.go
-	currentIncident.Lock()
-	defer currentIncident.Unlock()
-
-	if created {
-		currentIncident.StartedAt = ev.Time
-		err = currentIncident.Sync()
-		if err != nil {
-			_, _ = fmt.Fprintln(w, err)
-
-			l.logger.Errorln(err)
-			return
-		}
-
-		l.logger.Infof("[%s %s] opened incident", obj.DisplayName(), currentIncident.String())
-
-		historyRow := &incident.HistoryRow{
-			Type:    incident.Opened,
-			Time:    types.UnixMilli(ev.Time),
-			EventID: utils.ToDBInt(ev.ID),
-		}
-		_, err = currentIncident.AddHistory(historyRow, false)
-	}
-
-	err = currentIncident.AddEvent(l.db, &ev)
-	if err != nil {
-		l.logger.Errorln(err)
-		return
-	}
-
 	l.logger.Infof("processing event")
 
-	if ev.Type == event.TypeAcknowledgement {
-		err := l.ProcessAcknowledgementEvent(currentIncident, ev)
-		if err != nil {
-			_, _ = fmt.Fprintln(w, err)
-
-			l.logger.Errorln(err)
-		}
-
-		return
-	}
-
-	oldIncidentSeverity := currentIncident.Severity()
-
-	if currentIncident.SeverityBySource == nil {
-		currentIncident.SeverityBySource = make(map[int64]event.Severity)
-	}
-
-	oldSourceSeverity := currentIncident.SeverityBySource[ev.SourceId]
-	if oldSourceSeverity == event.SeverityNone {
-		oldSourceSeverity = event.SeverityOK
-	}
-
-	if oldSourceSeverity == ev.Severity {
-		msg := fmt.Sprintf("%s: ignoring superfluous %q state event from source %d", obj.DisplayName(), ev.Severity.String(), ev.SourceId)
-		_, _ = fmt.Fprintln(w, msg)
-
-		l.logger.Warnln(msg)
+	if err := currentIncident.ProcessEvent(ev, created); err != nil {
+		_, _ = fmt.Fprintln(w, err)
 		return
 	}
```

Comment:
> Are there any errors where you wouldn't want to log it?

No! But usually the logs contain more information about the errors than a simple http response. That's why I chose this way, because the individual functions that generate these errors can provide more details than here in listener and decide themselves which error they pass to the caller. The logging generally gets more naturell and nice with this [PR](https://github.com/Icinga/icinga-notifications/pull/76).

Reply:
My thinking was that here, the primary error reporting mechanism should be the log, not the HTTP response. If there's a problem with the database, this should always show up in our logs, the user shouldn't have to search logs of all sources for errors (if they'd even log them in a useful way).

Primarily, I noticed this because it is inconsistent with other error handling in the same function:
https://github.com/Icinga/icinga-notifications/blob/3a6912e66b108288387dabbe022f402fd417381d/internal/listener/listener.go#L116-L123

In general, I just want to ensure that every error (except those resulting in 400 Bad Request due to bad input) to show up in the log. If it was done here, it would be easier to verify (for me as part of the review), but sure, logging where the error first appears will likely result in more useful messages.